### DISCUSSION TITLE
Is There a Way to Restrict Who Can Create Servers?

### DISCUSSION THREAD

**Discussion by zicklag** [stoatchat/discussions#402] [Q&A] [answered]
I setup a self-hosted revolt server, but I would like to configure it so only an admin user is allowed to create new chat servers. Is there a way to do this?

I don't want just anybody being able to come on and use my server resources or domain to host their own chat servers about whatever they want.

**Comment by TheYellowArchitect** [comment]
Privacy-focused yet self-hosting seems an afterthought, sad. I would have instantly made the jump and self-host a server on my VPS.

XMPP and Matrix clients have shit UI, cannot even upload custom emojis.
But you can self-host. Here, it is the opposite, sigh, never a perfect middle. Eternal IRC chat it seems...

**Reply by insertish** [reply]
I don't have the resources to support self-hosted instances, I'm one developer accounting for most of the work done here.
That said, I am providing the necessary resources to run an instance anyways here: https://github.com/revoltchat/self-hosted

**Comment by insertish** [comment, marked as answer]
There's no way to do this directly from Revolt currently but you can block access to the `POST /servers/create` route in your reverse proxy.

**Reply by zicklag** [reply]
That's a good idea I wouldn't have thought of. :+1: That should work.

If I manage to find time, would you be open to a pull request that adds the ability to configure a whitelist of users who are allowed to create new servers?

**Reply by insertish** [reply]
Yeah that would be fine, I want to redo how configuration is loaded in the future but for the time being this can go into the 'delta' specific environment variables, there's a file in the `quark` crate for this. I think it's enough to just search for `delta.rs`.

Currently typing on phone the formatting might be off.
